src/Integration/Connection/UI/OrganizationSelectionWindow.xaml.cs

```diff
@@ -40,15 +43,29 @@ internal OrganizationSelectionWindow(IEnumerable<SonarQubeOrganization> organiza
             OrganizationComboBox.ItemsSource = sortedOrganizations;
         }
 
-        private void OnOkButtonClick(object sender, RoutedEventArgs e)
+        private void OnOwnOkButtonClick(object sender, RoutedEventArgs e)
```

Comment:
To keep things simple, I bound each `OK` button to a different handler method to fetch the data from the combobox/textbox as appropriate to work out which organization was selected.